node/src/accountant/dao_utils.rs

```diff
@@ -876,73 +858,120 @@ mod tests {
         let _ = subject.make_connection();
     }
 
+    fn create_table_with_text_id_and_single_numeric_column(
+        conn: &Connection,
+        init_data: &[(&str, i64)],
+    ) {
+        conn.execute("create table example (name text, num integer)", [])
+            .unwrap();
+        if !init_data.is_empty() {
+            let sequence_of_values_for_inserted_rows = init_data
+                .iter()
+                .map(|(name, num)| format!("({}, {})", name, num))
+                .join(", ");
+            let rows_added = conn
+                .execute(
+                    &format!(
+                        "insert into example (name, num) values {}",
+                        sequence_of_values_for_inserted_rows
+                    ),
+                    [],
+                )
+                .unwrap();
+            assert_eq!(rows_added, init_data.len())
+        }
+    }
+
+    const UPDATE_STM_WITH_RETURNING: &str = "update example set num = num + 2 returning num";
+
     #[test]
-    fn multi_update_rows_changed_returns_all_satisfying_results() {
-        let random_collection_of_changed_data = vec![Ok(5_i64), Ok(111), Ok(4321)];
-        let iterator = random_collection_of_changed_data
-            .into_iter()
-            .map(|res| match res {
-                Ok(num) => Ok(num > 0),
-                Err(e) => Err(e),
-            });
+    fn update_rows_and_return_their_count_returns_all_satisfying_results() {
+        let conn = Connection::open_in_memory().unwrap();
+        create_table_with_text_id_and_single_numeric_column(
+            &conn,
+            &vec![("'A'", 12), ("'B'", 23), ("'C'", 34)],
+        );
+        let mut returning_update_stm = conn.prepare(UPDATE_STM_WITH_RETURNING).unwrap();
+        let function_to_validate_row_value = |row: &Row| row.get::<usize, i64>(0).map(|_num| true);
```

Comment:
This closure would be clearer--that it always returns true, I mean--if it were made simpler.

Reply:
I've decided to decline here. I'd rather be sure that an error (yeah, poor hypothesis) did not fell out when we asked for the queried value returned at the row.